### PR TITLE
Adds docs for vbeta-0.3.0 features

### PR DESCRIPTION
**Pull request by srm09**

### CODE REVIEW

**Comment by srm09** [comment]
@vladimirvivien I added the go report card badge and some corrections for `gofmt`

File: README.md [hunk 1]
```diff
@@ -142,6 +163,25 @@ hosts=host_list_provider(hosts=["some.host", "172.100.100.20"], ssh_config=ssh)
...
+
+```bash
+# starts a ssh-agent
+$> eval "$(ssh-agent -s)"
+$> ssh-add /foo/bar_rsa
+...
+# ssh/scp connections in the diagnotics.crsh script will use the above ssh-agent
+$> crashd run diagnostics.crsh 
+```
+
```
Comment (vladimirvivien) [inline README.md:177]: Add a small sentence or two about using ssh-agent for passphrase-protected keys.

File: README.md [hunk 2]
```diff
@@ -142,6 +163,25 @@ hosts=host_list_provider(hosts=["some.host", "172.100.100.20"], ssh_config=ssh)
...
+If the crashd operator does not want to rely on the default ssh-agent process, the crashd configuration function (`crashd_config()`) has an option to start a new instance of the ssh-agent which will be used for all corresponding ssh/scp connections in the script.
```
Comment (vladimirvivien) [inline README.md:178]: Add new heading `Using Internal Crashd ssh-agent`

**Review by vladimirvivien — COMMENTED** [review]
document ssh-agent with new headings

File: README.md [hunk 3]
```diff
@@ -142,6 +163,38 @@ hosts=host_list_provider(hosts=["some.host", "172.100.100.20"], ssh_config=ssh)
 ...
 ```
 
+### SSH-Agent for ssh and scp connections
```
Comment (vladimirvivien) [inline README.md:166]: I don't think this makes sense here, let's move that block here - https://github.com/vmware-tanzu/crash-diagnostics/tree/main/docs#crashd_config at the end of that section (crashd_config).

**Review by vladimirvivien — COMMENTED** [review]
Quick change

**Review by vladimirvivien — APPROVED** [review]
LGTM

**Pull request merged**